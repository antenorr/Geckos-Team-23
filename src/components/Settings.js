import React, { Component } from 'react';
import { render } from "react-dom";

class Settings extends Component {
  constructor() {
    super();
    this.updateTabWindow = this.updateTabWindow.bind(this);
    this.updatePinnedTabs = this.updatePinnedTabs.bind(this);
    this.updateAutoStart = this.updateAutoStart.bind(this);
  }
  updateTabWindow = (e) => {
    if (e.target.checked) {
        this.props.updateTabWindow(e.target.value)
    }
  }

  updatePinnedTabs = (e) => {
    if (e.target.checked) {
        this.props.updatePinnedTabs(e.target.value)
    }
  }

  updateAutoStart = (e) => {
    if (e.target.checked) {
        this.props.updateAutoStart(e.target.value)
    }
  }

  render() {
    return (
      <div id="settings" className="main">
        <div id="openTabWindow">
          <form id="openTabWindow">
            <h4 className="settings-option">When opening tab(s), do so in:</h4>
            <div>
              <label className="settings-input">
                  <input
                    type="radio"
                    name="windowOptions"
                    value='currentWindow'
                    checked={this.props.tabWindow === "currentWindow"}
                    onChange={this.updateTabWindow} 
                    
                  /> Current Window
              </label>
            </div>
            <div>
              <label className="settings-input">
                  <input
                      type="radio"
                      name="windowOptions"
                      value='newWindow'
                      checked={this.props.tabWindow === "newWindow"}
                      onChange={this.updateTabWindow} 
                      
                    /> New Window
              </label>
            </div>
          </form>
        </div>

        <div id="pinnedTabs">
          <form id="pinnedTabs">
            <h4 className="settings-option">Pinned tabs are: </h4>
            <div>
              <label className="settings-input">
                <input
                    type="radio"
                    name="pinnedOptions"
                    value='noPinned'
                    checked={this.props.pinnedTab === "noPinned"}
                    onChange={this.updatePinnedTabs} 
                  /> Not saved
              </label>
            </div>
            <div>
              <label className="settings-input">
                <input
                    type="radio"
                    name="pinnedOptions"
                    value='yesPinned'
                    checked={this.props.pinnedTab === "yesPinned"}
                    onChange={this.updatePinnedTabs} 
                  /> Will be saved
              </label>
            </div>
          </form>
        </div>

        <div id="autoStartup">
          <form id="autoStartup">
              <h4 className="settings-option">Start OneTab++: </h4>
              <div>
                <label className="settings-input">
                  <input
                      type="radio"
                      name="pinnedOptions"
                      value='autoStartNo'
                      checked={this.props.autoStart === "autoStartNo"}
                      onChange={this.updateAutoStart} 
                    /> I will start it myself
                </label>
              </div>
              <div>
                <label className="settings-input">
                  <input
                      type="radio"
                      name="autoOptions"
                      value='autoStart'
                      checked={this.props.autoStart === "autoStart"}
                      onChange={this.updateAutoStart} 
                    /> Start it when I open my browser
                </label>
              </div>
          </form>
        </div>

        <div className="settings-div">
          <div className="btn-group  btn-center">
            <button 
              onClick={this.props.handleSettingsSubmit} 
              className="button settings-btn">Save</button>
          </div>
        </div>
      </div>
    );
  }
}
export default Settings;


