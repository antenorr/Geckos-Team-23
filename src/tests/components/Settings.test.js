import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import Settings from '../../components/Settings';

test('should render Settings correctly', () => {
  const wrapper = shallow(<Settings />);
  expect(toJSON(wrapper)).toMatchSnapshot();
});