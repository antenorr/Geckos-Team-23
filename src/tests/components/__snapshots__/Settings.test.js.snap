// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Settings correctly 1`] = `
<div
  className="main"
  id="settings"
>
  <div
    id="openTabWindow"
  >
    <form
      id="openTabWindow"
    >
      <h4
        className="settings-option"
      >
        When opening tab(s), do so in:
      </h4>
      <div>
        <label
          className="settings-input"
        >
          <input
            checked={false}
            name="windowOptions"
            onChange={[Function]}
            type="radio"
            value="currentWindow"
          />
           Current Window
        </label>
      </div>
      <div>
        <label
          className="settings-input"
        >
          <input
            checked={false}
            name="windowOptions"
            onChange={[Function]}
            type="radio"
            value="newWindow"
          />
           New Window
        </label>
      </div>
    </form>
  </div>
  <div
    id="pinnedTabs"
  >
    <form
      id="pinnedTabs"
    >
      <h4
        className="settings-option"
      >
        Pinned tabs are: 
      </h4>
      <div>
        <label
          className="settings-input"
        >
          <input
            checked={false}
            name="pinnedOptions"
            onChange={[Function]}
            type="radio"
            value="noPinned"
          />
           Not saved
        </label>
      </div>
      <div>
        <label
          className="settings-input"
        >
          <input
            checked={false}
            name="pinnedOptions"
            onChange={[Function]}
            type="radio"
            value="yesPinned"
          />
           Will be saved
        </label>
      </div>
    </form>
  </div>
  <div
    id="autoStartup"
  >
    <form
      id="autoStartup"
    >
      <h4
        className="settings-option"
      >
        Start OneTab++: 
      </h4>
      <div>
        <label
          className="settings-input"
        >
          <input
            checked={false}
            name="pinnedOptions"
            onChange={[Function]}
            type="radio"
            value="autoStartNo"
          />
           I will start it myself
        </label>
      </div>
      <div>
        <label
          className="settings-input"
        >
          <input
            checked={false}
            name="autoOptions"
            onChange={[Function]}
            type="radio"
            value="autoStart"
          />
           Start it when I open my browser
        </label>
      </div>
    </form>
  </div>
  <div
    className="settings-div"
  >
    <div
      className="btn-group  btn-center"
    >
      <button
        className="button settings-btn"
      >
        Save
      </button>
    </div>
  </div>
</div>
`;
